data/mods/Magiclysm/Spells/black_dragon.json

```diff
@@ -0,0 +1,236 @@
+    "base_casting_time": 200,
+    "base_energy_cost": 250,
+    "min_duration": 1000,
+    "max_duration": 5000,
+    "duration_increment": 1000
+  },
+  {
+    "id": "dragon_shell_retaliation_black",
+    "type": "SPELL",
+    "name": "Black Dragon Shell Retaliation",
+    "description": "Acid burst that punishes attackers.",
+    "effect": "target_attack",
+    "effect_str": "dragon_acidburn",
+    "affected_body_parts": [ "torso" ],
+    "valid_targets": [ "hostile" ],
+    "min_damage": 0,
+    "max_damage": 10,
+    "damage_increment": 1.0,
+    "flags": [ "RANDOM_DAMAGE", "RANDOM_AOE" ],
+    "min_aoe": 0,
+    "max_aoe": 3,
+    "aoe_increment": 1.0,
+    "min_range": 1,
+    "max_range": 1,
+    "spell_class": "NONE",
+    "damage_type": "acid"
+  },
+  {
+    "id": "dragon_terror",
+    "type": "SPELL",
+    "name": "Dragon Terror",
+    "description": "With an ear piercing roar, you channel the inner power of dragons and strike fear into those who stand before you, slowing their escape from your terror.",
+    "valid_targets": [ "hostile", "ground" ],
+    "effect": "target_attack",
+    "effect_str": "dragon_terror_debuff",
+    "affected_body_parts": [ "torso" ],
+    "max_level": 5,
+    "min_damage": 0,
+    "max_damage": 0,
+    "min_aoe": 2,
+    "max_aoe": 10,
+    "aoe_increment": 2,
+    "min_range": 1,
+    "max_range": 1,
+    "min_duration": 1000,
+    "max_duration": 5000,
+    "duration_increment": 1000,
+    "base_casting_time": 200,
+    "base_energy_cost": 400,
+    "energy_source": "MANA",
+    "difficulty": 0,
+    "flags": [ "LOUD", "NO_HANDS", "NO_LEGS" ]
+  },
+  {
+    "id": "dragon_boost",
+    "type": "SPELL",
+    "name": "Dragon Boost",
+    "description": "You let out a mighty roar as you infuse yourself with the power of Dragons.",
+    "valid_targets": [ "self" ],
+    "effect": "target_attack",
+    "effect_str": "dragon_boost",
+    "extra_effects": [ { "id": "health_recovery", "hit_self": true }, { "id": "stamina_recovery", "hit_self": true } ],
+    "max_level": 5,
+    "min_range": 1,
+    "max_range": 1,
+    "min_duration": 2000,
+    "max_duration": 10000,
+    "duration_increment": 2000,
+    "base_casting_time": 100,
+    "base_energy_cost": 150,
+    "energy_source": "MANA",
+    "difficulty": 0,
+    "flags": [ "LOUD", "NO_HANDS", "NO_LEGS" ]
+  },
+  {
+    "type": "SPELL",
+    "id": "health_recovery",
+    "name": "Health recovery",
+    "description": "minor health recovery spell associated with Dragon Boost.",
+    "valid_targets": [ "self" ],
+    "effect": "target_attack",
+    "min_damage": 0,
+    "max_damage": -10,
+    "flags": [ "SILENT", "RANDOM_DAMAGE" ]
+  },
+  {
+    "type": "SPELL",
+    "id": "stamina_recovery",
+    "name": "Stamina recovery",
+    "description": "minor stamina recovery spell associated with Dragon Boost.",
+    "valid_targets": [ "self" ],
+    "effect": "recover_energy",
+    "effect_str": "STAMINA",
+    "min_damage": 0,
+    "max_damage": -10,
```

Comment:
max stamina is currently 10,000 so an addition of 10 is really small. what kind of % were you intending here?

Reply:
Not a lot, no more than what 10 HP would be so it's balanced with the health healing spell. I guess that would put it somewhere around 150 or so?

Reply:
You should use a positive value here, if you want to add to stamina reserves. A value of 1000 would be 10% of total stamina. See this test case on spell recovery for example:

https://github.com/CleverRaven/Cataclysm-DDA/blob/2345fc2470f7d4ff5d011676f3f925ebfe88b724/tests/magic_spell_test.cpp#L632-L635